File: TeamService/Models/Team.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TeamService.Models;

[Table("team", Schema = "public")]
public class Team
{
    [Key]
    [Column("teamid")]
    public int TeamId { get; set; }

    [Column("professorid")]
    [Required]
    public Guid ProfessorId { get; set; }

    [Column("teamname")]
    [Required]
    [MaxLength(48)]
    public string TeamName { get; set; } = string.Empty;

    [Column("description")]
    [MaxLength(128)]
    public string? Description { get; set; }

    [Column("teampic")]
    [MaxLength(255)]
    public string? TeamPic { get; set; }

    [Column("createdat")]
    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<TeamMembership>? TeamMemberships { get; set; }
    public ICollection<TeamAsset>? TeamAssets { get; set; }
    public MarketConfiguration? MarketConfiguration { get; set; }
}

File: TeamService/app/models/team.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
import uuid 

from app.database import Base

class Team(Base):
    __tablename__ = "teams"

    team_id = Column("teamid", Integer, primary_key=True, index=True, autoincrement=True)
    
    public_id = Column("publicid", UUID(as_uuid=True), default=uuid.uuid4, unique=True, nullable=False)
    
    professor_id = Column("professorid", UUID(as_uuid=True), nullable=False)
    name = Column("teamname", String(48), nullable=False)
    description = Column("description", String(128), nullable=True)
    team_pic = Column("teampic", String(255), nullable=True)
    
    access_code = Column("accesscode", String(20), unique=True, nullable=False)
    
    created_at = Column("createdat", DateTime(timezone=True), server_default=func.now())

    market_configuration = relationship(
        "MarketConfiguration",
        uselist=False,
        back_populates="team",
        cascade="all, delete-orphan"
    )

    memberships = relationship(
        "TeamMembership",
        back_populates="team",
        cascade="all, delete"
    )

    team_assets = relationship(
        "TeamAsset",
        back_populates="team",
        cascade="all, delete"
    )
